src/linux/namespaces.rs

```diff
@@ -475,3 +524,10 @@ fn normalize_path(path: &Path) -> PathBuf {
 
     normalized
 }
+
+/// Check if a path contains any symlinks.
+fn path_has_symlinks(path: &Path) -> bool {
+    path.ancestors()
+        .flat_map(|path| path.symlink_metadata().ok())
+        .any(|metadata| metadata.is_symlink())
```

Comment:
Hmm... Is there a reason that you've switched to `.symlink_metadata()` instead of `.read_link()`?

I think they should be equivalent. Just curious if there is a case I'm not thinking of.

Maybe the `lstat` sycall is more efficient than `readlink` because it doesn't read the contents of the symlink?